client/asset/dcr/dcr.go

```diff
@@ -5117,6 +5159,123 @@ func (dcr *ExchangeWallet) EstimateSendTxFee(address string, sendAmount, feeRate
 	return finalFee, isValidAddress, nil
 }
 
+func (dcr *ExchangeWallet) isSPV() bool {
+	return dcr.walletType == walletTypeSPV
+}
+
+func (dcr *ExchangeWallet) StakeStatus() (*asset.TicketStakingStatus, error) {
+	if !dcr.connected.Load() {
+		return nil, errors.New("not connected, login first")
+	}
+	sdiff, err := dcr.wallet.StakeDiff(dcr.ctx)
+	if err != nil {
+		return nil, err
+	}
+	isRPC := !dcr.isSPV()
+	var vspURL string
+	if !isRPC {
+		if v := dcr.vspV.Load(); v != nil {
+			vspURL = v.(*vsp).URL
+		}
+	}
+	tickets, err := dcr.wallet.Tickets(dcr.ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving tickets: %w", err)
+	}
+	voteChoices, tSpendPolicy, treasuryPolicy, err := dcr.wallet.VotingPreferences(dcr.ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving stances: %w", err)
+	}
+	return &asset.TicketStakingStatus{
+		TicketPrice: uint64(sdiff),
+		VSP:         vspURL,
+		IsRPC:       isRPC,
+		Tickets:     tickets,
+		Stances: asset.Stances{
+			VoteChoices:    voteChoices,
+			TSpendPolicy:   tSpendPolicy,
+			TreasuryPolicy: treasuryPolicy,
+		},
+	}, nil
+}
+
+func vspInfo(url string) (*vspdjson.VspInfoResponse, error) {
+	suffix := "/api/v3/vspinfo"
+	path, err := neturl.JoinPath(url, suffix)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := http.Get(path)
+	if err != nil {
+		return nil, fmt.Errorf("http get error: %v", err)
+	}
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	var info vspdjson.VspInfoResponse
+	err = json.Unmarshal(b, &info)
+	if err != nil {
+		return nil, err
+	}
+	return &info, nil
+}
+
+// SetVSP sets the VSP provider. Ability to set should be checked with CanSetVSP
+// first. Part of the asset.TicketBuyer interface.
```

Comment:
I can't find `CanSetVSP` anywhere in the code. Yet to implement?

Reply:
The ability to detect if we can set was added to the returns from `StakeStatus()` `IsRPC` Will remove from comment.

Reply:
Changed.